stl/inc/xstring

```diff
@@ -82,13 +93,25 @@ struct _Char_traits { // properties of a string or stream element
         return nullptr;
     }
 
-    static _Elem* move(_Out_writes_all_(_Count) _Elem* const _First1, _In_reads_(_Count) const _Elem* _First2,
-        size_t _Count) noexcept /* strengthened */ {
+    static _CONSTEXPR20 _Elem* move(_Out_writes_all_(_Count) _Elem* const _First1,
+        _In_reads_(_Count) const _Elem* _First2, const size_t _Count) noexcept /* strengthened */ {
         // copy [_First2, _First2 + _Count) to [_First1, ...), allowing overlap
-        return static_cast<_Elem*>(_CSTD memmove(_First1, _First2, _Count * sizeof(_Elem)));
+#ifdef __cpp_lib_is_constant_evaluated
+        if (!_STD is_constant_evaluated())
+#endif // __cpp_lib_is_constant_evaluated
+        {
+            return static_cast<_Elem*>(_CSTD memmove(_First1, _First2, _Count * sizeof(_Elem)));
+        }
+
+        for (size_t _Idx = 0; _Idx < _Count; ++_Idx) {
+            *(_First1 + _Idx) = *(_First2 + _Idx);
+        }

```

Comment:
This won't work if `_First1` is in `[_First2, _First2 + _Count)`, since it overwrites the contents of `[_First1, _First2 + _Count)` before copying from that range.
```suggestion
        if (_First1 <= _First2) {
            for (size_t _Idx = 0; _Idx < _Count; ++_Idx) {
                *(_First1 + _Idx) = *(_First2 + _Idx);
            }
        } else {
            for (size_t _Idx = _Count; _Idx-- > 0;) {
                *(_First1 + _Idx) = *(_First2 + _Idx);
            }
        }
```

Reply:
Any reason to not copy backwards by default?

Reply:
Also related. All other siblings of this method have a non-`const` `_Count` parameter that is then directly decremented in the loop. I believe we should stay consistent and follow the same pattern everywhere:

```cpp
    _NODISCARD static _CONSTEXPR17 const _Elem* find(
        _In_reads_(_Count) const _Elem* _First, size_t _Count, const _Elem& _Ch) noexcept /* strengthened */ {
        // look for _Ch in [_First, _First + _Count)
        for (; 0 < _Count; --_Count, ++_First) {
            if (*_First == _Ch) {
                return _First;
            }
        }

        return nullptr;
    }
```

Reply:
> Any reason to not copy backwards by default?

Copying backwards doesn't work if `_First2` is in `[_First1, _First1 + _Count)`.

> All other siblings of this method have a non-`const` `_Count` parameter that is then directly decremented in the loop. I believe we should stay consistent and follow the same pattern everywhere.

I have no strong feelings, so feel free to make changes if you do.